apps/api-gateway/src/core/celery_tasks.py

```
@@ -1354,3 +1354,173 @@ def _write_xlsx(file_path: str, headers: list, rows: list) -> int:
+    import json
+    import tarfile
+    import tempfile
+    from datetime import datetime, timezone
+
+    async def _run():
+        from sqlalchemy.ext.asyncio import create_async_engine, AsyncSession
+        from sqlalchemy.orm import sessionmaker
+        from sqlalchemy import text
+
+        db_url = os.getenv("DATABASE_URL", "postgresql+asyncpg://postgres:postgres@localhost/zhilian")
+        backup_dir = os.getenv("BACKUP_TMP_DIR", "/tmp/backups")
```

Ruff (0.15.2)

[error] Probable insecure usage of temporary file or directory: "/tmp/backups"

(S108)